services/web/server/src/simcore_service_webserver/cli.py

```diff
@@ -114,4 +117,26 @@ def main(args: Optional[List] = None):
     setup_logging(level=config["main"]["log_level"], slow_duration=slow_duration)
 
     # run
-    run_service(config)
+    app = create_application(config)
+    return (app, config)
+
+
+def main(args: Optional[List] = None):
+    app, config = _setup_app(args)
+    run_service(app, config)
+
+
+async def app_factory() -> web.Application:
+    # parse & config file
+    app_settings = ApplicationSettings()
+    assert app_settings.build_target  # nosec
+    log.info("Application settings: %s", f"{app_settings.json(indent=2)}")
+    args = [
+        "--config",
+        "server-docker-dev.yaml"
+        if app_settings.build_target.lower() == "development"
+        else "server-docker-prod.yaml",
+    ]
+    app, _ = _setup_app(args)
```

Comment:
MINOR: 
1. use ``models_library.basic_types.BootModeEnum.DEVELOPMENT`` enum instead of string (actually we should extend this to all repo)
2. compact code
3. further cleanup will happen when PR #2376 is ready

```python
from models_library.basic_types import BootModeEnum

ConfigDict = Dict[str, Any]

def _setup_app( *cmd_args ) -> Tuple[web.Application, ConfigDict]
# ... 

async def app_factory() -> web.Application:
   # ...
   app, _ = _setup_app(
        "--config",
        "server-docker-dev.yaml"
        if app_settings.build_target == BootModeEnum.DEVELOPMENT
        else "server-docker-prod.yaml",
    )


```

Reply:
well I agree with that one, but I did not want to start removing trafaret in this PR. I try and hope I don't destroy something else 💪